public/app/features/dashboard/components/PanelEditor/state/actions.ts

```diff
@@ -144,6 +145,15 @@ export function exitPanelEditor(): ThunkResult<void> {
       }, 20);
     }
 
+    // A new panel is only new until the first time we exit the panel editor
+    if (sourcePanel.isNew) {
+      if (!shouldDiscardChanges) {
+        delete sourcePanel.isNew;
+      } else {
+        dashboard && removePanel(dashboard, sourcePanel, true);
+      }
+    }
```

Comment:
This is one way of handling the new panel: adding a prop to the model and looking for it when leaving the editing page (because `Apply` and going back would not remove the property from the model). 

Another way would be to let the Panel Edit page know whether its panel is new (with an "`isNew`" URL param e.g.) and let it call `removePanel()` on _discard_. That way we wouldn't have to worry about `Apply` and "going back manually" because only the `Add` buttons would trigger the URL with `isNew` param.